arc/src/main.c

```diff
@@ -667,30 +781,36 @@ static void handle_sensor(struct zjs_ipm_message* msg)
                 error_code = ERROR_IPM_OPERATION_FAILED;
                 ZJS_PRINT("failed to initialize BMI160 sensor\n");
             } else {
-                if (!auto_calibration(bmi160)) {
+                if (auto_calibration(bmi160) != 0) {
```

Comment:
You don't really need the != 0, but that's just my personal preferred style, not going to enforce it. :)